openlibrary/core/lending.py

```diff
@@ -276,20 +281,23 @@ def get_realtime_availability_of_ocaid(ocaid):
 
 @public
 def add_availability(editions):
-    """Adds API v2 availability info to editions, e.g. for work's editions table
+    """
+    Adds API v2 'availability' key to editions, e.g. for work's editions table
+    :param list of dict editions:
+    :rtype: list of dict
     """
     def get_ocaid(ed):
         if ed.get('ocaid') or ed.get('identifier') or ed.get('ia'):
             return ed.get('ocaid') or ed.get('identifier') or (
-                ed.ia[0] if isinstance(ed.ia, list) else ed.ia)
+                ed['ia'][0] if isinstance(ed['ia'], list) else ed['ia'])
 
-    ocaids = [ocaid for ocaid in [get_ocaid(ed) for ed in editions] if ocaid]
-    availability = get_availability_of_ocaids(ocaids)
-    for i, ed in enumerate(editions):
+    ocaids = [get_ocaid(ed) for ed in editions]
+    ocaids = [ocaid for ocaid in ocaids if ocaid]
+    availabilities = get_availability_of_ocaids(ocaids)
+    for ed in editions:
         ocaid = get_ocaid(ed)
-        editions[i]['availability'] = availability.get(ocaid) if ocaid else {
-            'status': 'error'
-        }
+        errored = (ocaid is None) or (availabilities.get(ocaid) is None)
```

Comment:
What about other non-truthy values like "" ? Suggest the following instead as both more direct and concise:

```
success = ocaid and availabilities.get(ocaid)
... if success else {'status': 'error'}
```